scripts/templates/generate_coverage_report

```diff
@@ -130,19 +131,35 @@ if [[ "$1" == "--codecov" ]]; then
    remove_gcda_files
 
    # Upload report to CodeCov
+   if ! [ -d "$BUILD/tmp" ]; then
+      mkdir -p "$BUILD/tmp"
+   fi
 
-   if ! bash <(curl -s https://codecov.io/bash) -X gcov; then
+   pushd "$BUILD/tmp"
 
-      echo "ERROR: codecov did not collect coverage reports"
-      echo "Trying again in 1 minute..."
-      sleep 60
+   if ! [ -f codecov ]; then
+
+      curl -Os https://uploader.codecov.io/latest/linux/codecov
+      if ! [ -f codecov ]; then
+         echo "ERROR: codecov did not collect coverage reports"
+         echo "Trying again in 1 minute..."
+         sleep 60
+      fi
 
-      if ! bash <(curl -s https://codecov.io/bash) -X gcov; then
+      curl -Os https://uploader.codecov.io/latest/linux/codecov
```

Comment:
**BUG:** this will run unconditionally, even if the first curl succeeded!
There was a reason why that was under an `if` in the old code.
Please fix it.